### PR TITLE
[CORE] Replaces Pair class

### PR DESCRIPTION
**Pull request by srossbach**
Thrown away all the fancy code that no one will ever use.

### CODE REVIEW

**Comment by m273d15** [comment]
Fixes issue #310

**Comment by srossbach** [comment]
I guess we can use the Apache Version. I will push another update so if you introduce the library you only have to change the imports.

As for the LineRange, I highly disagree as this is an object that is being marshalled. I am just wondering why there are no XStream annotations. I guess we should configure XStream in such a way that is does not marshall any class that is not annotated.

File: de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/util/CollaborationUtils.java [hunk 1]
```diff
@@ -256,38 +256,35 @@ public void run() {
...
-            String.format(
-                "\nProject: %s, Files: %d, Size: %s",
-                project.getName(), fileCountAndSize.v, format(fileCountAndSize.p)));
-      } else {
-        List<IResource> resources =
-            ResourceAdapterFactory.convertBack(sarosSession.getSharedResources(project));
+      final List<IResource> resources;
 
-        fileCountAndSize = FileUtils.getFileCountAndSize(resources, false, IResource.NONE);
+      if (isCompletelyShared)
```
Comment (donut87) [inline de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/util/CollaborationUtils.java]: Inline this, if you got rid of the ternary operator please.

File: de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/util/CollaborationUtils.java [hunk 2]
```diff
@@ -256,38 +256,35 @@ public void run() {
...
+          FileUtils.getFileCountAndSize(
+              resources,
+              isCompletelyShared ? true : false,
+              isCompletelyShared ? IContainer.EXCLUDE_DERIVED : IResource.NONE);
+
+      result.append(
+          String.format(
+              "\nProject: %s (%s), Files: %d, Size: %s",
+              project.getName(),
+              isCompletelyShared ? "complete" : "partial",
```
[the remark this answers is not in the document]
Comment (srossbach) [reply de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/util/CollaborationUtils.java]: Benefit ?

[the remark this answers is not in the document]
Comment (donut87) [reply de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/util/CollaborationUtils.java]: Readability as always. The consequences resulting from the same fact are 'cluttered' all over the place as opposed to bundled in one block. I do not like having 4 if statements if one would suffice.

**Comment by m273d15** [comment]
@srossbach Why did you merged the PR without addressing the comments of @donut87 ?

**Comment by srossbach** [comment]
Because most of the code in this class is:

1. Duplicated (IntelliJ has also most of the code)
2. Most of the code will be deleted anyway when the HTML
3. Most of the code in this class needs an overhaul anyway

**Comment by donut87** [comment]
And you couldn't have written this before merging? Or even before asking me about the benefit?
I get your point and if this code is going to be worked on then fine, leave it that way and maybe add a TODO for the upcoming clean up work, but not reacting on a comment is just plain rude.

**Pull request merged**